lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -690,14 +690,17 @@ def __init__(
         super().__init__(charm, relation_name)
         self.charm = charm
         self._relation_name = relation_name
+        self.container = self.charm.unit.get_container("loki")
 
         # If Loki is run in single-tenant mode, all the chunks are put in a folder named "fake"
         # https://grafana.com/docs/loki/latest/operations/storage/filesystem/
         # https://grafana.com/docs/loki/latest/rules/#ruler-storage
         tenant_id = "fake"
         self._rules_dir = os.path.join(rules_dir, tenant_id)
+        # create tenant dir so that the /loki/api/v1/rules endpoint returns "no rule groups found"
+        # instead of "unable to read rule dir /loki/rules/fake: no such file or directory"
+        self.container.make_dir(self._rules_dir, make_parents=True)
```

Comment:
Same here, you can use:

```python
self.charm._container.make_dir(self._rules_dir, make_parents=True)
```

Reply:
Changed about to set `self.container` - loki lib code currently uses `self.container` in several places.